src/containers/PaymentMethodsPage/PaymentMethodsPage.duck.js

```diff
@@ -1 +1,95 @@
-//TODO
+import { fetchCurrentUser } from '../../ducks/user.duck';
+import { storableError } from '../../util/errors';
+import * as log from '../../util/log';
+
+// ================ Action types ================ //
+
+export const SETUP_INTENT_REQUEST = 'app/CheckoutPage/SETUP_INTENT_REQUEST';
+export const SETUP_INTENT_SUCCESS = 'app/CheckoutPage/SETUP_INTENT_SUCCESS';
+export const SETUP_INTENT_ERROR = 'app/CheckoutPage/SETUP_INTENT_ERROR';
+
+export const STRIPE_CUSTOMER_REQUEST = 'app/CheckoutPage/STRIPE_CUSTOMER_REQUEST';
+export const STRIPE_CUSTOMER_SUCCESS = 'app/CheckoutPage/STRIPE_CUSTOMER_SUCCESS';
+export const STRIPE_CUSTOMER_ERROR = 'app/CheckoutPage/STRIPE_CUSTOMER_ERROR';
+
+// ================ Reducer ================ //
+
+const initialState = {
+  setupIntentInProgress: null,
+  setupIntentError: null,
+  stripeCustomerFetched: false,
+};
+
+export default function payoutMethodsPageReducer(state = initialState, action = {}) {
+  const { type, payload } = action;
+  switch (type) {
+    case SETUP_INTENT_REQUEST:
+      return { ...state, setupIntentInProgress: true, setupIntentError: null };
+    case SETUP_INTENT_SUCCESS:
+      return { ...state, setupIntentInProgress: null, setupIntentError: null };
+    case SETUP_INTENT_ERROR:
+      console.error(payload); // eslint-disable-line no-console
+      return { ...state, setupIntentInProgress: false, setupIntentError: null };
+    case STRIPE_CUSTOMER_REQUEST:
+      return { ...state, stripeCustomerFetched: false };
+    case STRIPE_CUSTOMER_SUCCESS:
+      return { ...state, stripeCustomerFetched: true };
+    case STRIPE_CUSTOMER_ERROR:
+      console.error(payload); // eslint-disable-line no-console
+      return { ...state, stripeCustomerFetchError: payload };
+    default:
+      return state;
+  }
+}
+
+// ================ Action creators ================ //
+
+export const setupIntentRequest = () => ({ type: SETUP_INTENT_REQUEST });
+export const setupIntentSuccess = () => ({ type: SETUP_INTENT_SUCCESS });
+export const setupIntentError = e => ({
+  type: SETUP_INTENT_ERROR,
+  error: true,
+  payload: e,
+});
+
+export const stripeCustomerRequest = () => ({ type: STRIPE_CUSTOMER_REQUEST });
+export const stripeCustomerSuccess = () => ({ type: STRIPE_CUSTOMER_SUCCESS });
+export const stripeCustomerError = e => ({
+  type: STRIPE_CUSTOMER_ERROR,
+  error: true,
+  payload: e,
+});
+// ================ Thunks ================ //
+
+export const createStripeSetupIntent = () => (dispatch, getState, sdk) => {
+  dispatch(setupIntentRequest());
+  return sdk.stripeSetupIntents
+    .create()
+    .then(response => {
+      dispatch(setupIntentSuccess());
+      const paymentIntent = response.data.data;
+      return paymentIntent;
```

Comment:
paymentIntent? setupIntent maybe.

Reply:
Is there any need to reference created setupIntent later? I.e. should it be added to store too `setupIntentSuccess(setupIntent)` ?